### PR TITLE
Fixed the RecordMapComponent - was not displaying pins.

### PR DESCRIPTION
**Pull request by chungvl**

### CODE REVIEW

File: src/components/records-map/records-map.ts [hunk 1]
```diff
@@ -44,7 +44,10 @@ export class RecordsMapComponent implements OnInit {
                 'zoom': 3.5,
             }
         });
-        this.loadMarkers();
+        if (this.navParams.data.hasOwnProperty('data')) {
+            this.records = this.navParams.get('data');
+        }
+        this.events.subscribe('home-willenter', () => timer(500).subscribe((foo) => this.ionViewWillEnter()));
```
Comment (tony-gaia) [inline src/components/records-map/records-map.ts]: Ok, so isn't ionViewWillEnter called automatically?

Assuming you do want to delay what occurs in ionViewWillEnter, put the timer function inside that method.

Also, no bogus variable names like ```foo``` - in this case you could just leave it blank

File: src/components/records-map/records-map.ts [hunk 2]
```diff
@@ -59,15 +62,31 @@ export class RecordsMapComponent implements OnInit {
                     if (record.hasOwnProperty('data') &&
                         record.data.hasOwnProperty('Latitude') &&
                         record.data.hasOwnProperty('Longitude')) {
-                        this.map.addMarkerSync({
-                            title: record.data['First Date'],
-                            icon: record.valid ? 'green' : 'blue',
+                        let title = record.data['Site ID'];
```
Comment (tony-gaia) [inline src/components/records-map/records-map.ts]: you can be fancy with string template literals and do something like:

```typescript
const title = `${record.data['Site ID']}${record.data['First Date'] ? record.data['First Date'] : ''}`;
```

File: src/components/records-map/records-map.ts [hunk 3]
```diff
@@ -59,15 +62,31 @@ export class RecordsMapComponent implements OnInit {
...
+                        if (record.data['First Date']) {
+                            title += ` ${record.data['First Date']}`;
+                        }
+                        const snippet = record.client_id || '';
+                        const marker = this.map.addMarkerSync({
+                            snippet: snippet,
+                            title: title,
+                            icon: record.valid ? '#ebffef' : '#ebf6ff',
```
Comment (tony-gaia) [inline src/components/records-map/records-map.ts]: use the consts.ts to make these colours effectively global consts

File: src/components/records-map/records-map.ts (hunk 2 above)
Comment (chungvl) [reply src/components/records-map/records-map.ts]: Yeah - my main issue with that one is undefined.toString() === 'undefined'

**Pull request merged**